spec/classes/foreman_proxy_content__container_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'spec_helper'
+
+describe 'foreman_proxy_content::container' do
+  on_supported_os.each do |os, facts|
+    context "on #{os}" do
+      let(:facts) { facts }
+      let(:pre_condition) do
+        <<-PUPPET
+        class { 'foreman_proxy':
+          foreman_ssl_cert => "/etc/foreman-proxy/foreman_ssl_cert.pem",
+          foreman_ssl_key => "/etc/foreman-proxy/foreman_ssl_key.pem",
+        }
+        include foreman_proxy_content
+        PUPPET
+      end
+
+      describe 'with inherited parameters' do
+
+        it { is_expected.to compile.with_all_deps }
+        it do
+          is_expected.to contain_apache__vhost__fragment('pulp-https-container')
+            .with_vhost('pulpcore-https')
+            .with_priority('10')
+            .with_content(%r{^\s+<Location "/pulpcore_registry/v2/">$})
+            .with_content(%r{^\s+Require expr %\{SSL_CLIENT_S_DN_CN\} == "foo.example.com"$})
+            .with_content(%r{^\s+RequestHeader set SSL_CLIENT_S_DN "admin"$})
+            .with_content(%r{^\s+</Location>$})
+            .with_content(%r{^\s+ProxyPass /v1/ https://foo\.example\.com:8443/container_gateway/v1/$})
+            .with_content(%r{^\s+ProxyPass /v2/ https://foo\.example\.com:8443/container_gateway/v2/$})
+            .with_content(%r{^\s+ProxyPassReverse /v1/ https://foo\.example\.com:8443/container_gateway/v1/$})
+            .with_content(%r{^\s+ProxyPassReverse /v2/ https://foo\.example\.com:8443/container_gateway/v2/$})
+            .with_content(%r{^\s+SSLProxyCheckPeerCN off$})
+            .with_content(%r{^\s+SSLProxyCheckPeerName off$})
+        end
+      end
+
+      describe 'with explicit parameters' do
+        let(:params) { { location_prefix: '/different_pulpcore_registry', registry_v1_path: '/vr1', registry_v2_path: '/vr2' } }
+
+        it { is_expected.to compile.with_all_deps }
```

Comment:
Seeing this error locally for some reason:
```
Failure/Error: it { is_expected.to compile.with_all_deps }
       error during compilation: Evaluation Error: Error while evaluating a Resource Statement, Duplicate declaration: Class[Foreman_proxy_content::Container] is already declared; cannot redeclare (line: 8) (line: 8, column: 1) on node cannolo.usersys.redhat.com
     # ./spec/classes/foreman_proxy_content__container_spec.rb:40:in `block (5 levels) in <top (required)>'
```

Not sure yet where this "Duplicate declaration" is happening.